### PR TITLE
fix(startup): only send one default_colors_set event during startup

### PR DESCRIPTION
**Pull request by bfredl**
related #26372

### CODE REVIEW

**Comment by echasnovski** [comment]
Thank you so much, @bfredl!

It does seem like it fixes the culprit of #26372, i.e. with `nvim --clean -u NORC --cmd 'call system(["sleep", "1"])' --cmd 'color blue'` background stays the same as it was before. And adding a `redraw` before sleep does show the proper data.

One thing I noticed is that setting `termguicolors` results into showing `#000000` background for me. I.e. `nvim --clean -u NORC --cmd 'set termguicolors' --cmd 'call system(["sleep", "1"])'` will show black background for a second and then show the new default color scheme. It would be good to make it work as `notermguicolors`, but I can see it being in the same realm as `redraw` for this use case.

**Comment by lewis6991** [comment]
Can confirm this avoids the flicker for me.

File: src/nvim/ui.c [hunk 1]
```diff
@@ -283,8 +284,21 @@ void ui_schedule_refresh(void)
...
+    ui_may_set_default_colors();
+  }
+}
+
+static void ui_may_set_default_colors(void)
+{
+  if (pending_default_colors) {
+    ui_call_default_colors_set(normal_fg, normal_bg, normal_sp,
+                               cterm_normal_fg_color, cterm_normal_bg_color);
+    pending_default_colors = false;
```
Comment (justinmk) [inline src/nvim/ui.c:300]:
```suggestion
    pending_default_colors = false;
    ui_call_default_colors_set(normal_fg, normal_bg, normal_sp,
                               cterm_normal_fg_color, cterm_normal_bg_color);
```

**Comment by echasnovski** [comment]
I've just retried it and it doesn't seem to work well with `termguicolors` (like in [this comment](https://github.com/neovim/neovim/pull/26381#issuecomment-1838192503)). With `termguicolors` becoming default for terminal emulators that support it, this isn't really fixing #26372. Am I missing something?

**Comment by bfredl** [comment]
I just chose to take one step at a time as I wasn't sure if termguicolors by default was decided yet, and this by itself fixes the issue for a lot of users so no reason to delay this PR. 

Will do a similar fix for the termguicolors event as a follow up.

**Comment by telemachus** [comment]
I'm not sure whether this is expected, but the flash still appears when I use neovim for man pages. (I.e., `MANPAGER=nvim +Man!`.)

I mention it only because I thought this fix removed the flash both when starting neovim normally and when running, e.g., `man nvim`, but now the flash is back for man pages.

**Comment by gpanders** [comment]
@telemachus You're seeing the same thing that @echasnovski [pointed out](https://github.com/neovim/neovim/pull/26381#issuecomment-1843533157). You can reproduce this with `nvim --cmd 'sleep 1'`. The `:Man` command does some nontrivial pre-processing which takes just long enough to make the "flash" visible. @bfredl already mentioned that a fix for `'termguicolors'` will be a follow up.

**Comment by gpanders** [comment]
@telemachus Can you try testing again on master with https://github.com/neovim/neovim/pull/26579 to see if you can still reproduce the issue?

**Comment by telemachus** [comment]
@gpanders Thanks for the ping. Unfortunately, I still have the same issue.

https://github.com/neovim/neovim/assets/62958/fffb9dd8-d209-4866-ae62-0ba6742158e1

In case it helps, here are some details.

+ nvim: NVIM v0.10.0-dev-1855+g0d26d192d
+ colorscheme: 
+ terminal: Ghostty cc6270f
+ os: macOS 14.2
+ Relevant (maybe?) bit of my `init.lua`:

    ```lua
    opt.termguicolors = true
    opt.background = "light"
    
    -- https://github.com/telemachus/github-nvim-theme
    local theme_loaded = safe_setup("github-theme", {
        theme_style = "light",
        sidebars = { "qf", "lf" },
        comment_style = "NONE",
        keyword_style = "NONE",
        function_style = "NONE",
        variable_style = "NONE",
        hide_end_of_buffer = false,
        hide_inactive_statusline = false,
        dark_float = true,
        dark_sidebar = true,
        transparent = true,
        overrides = function(c)
            return {
                Conceal = { fg = c.fg },
            }
        end,
    })
    if theme_loaded then
        cmd("colorscheme github_light")
    else
        cmd("colorscheme off")
    end
    ```

**Comment by gpanders** [comment]
What commit is 0d26d192d?

```
$ git show 0d26d192d
fatal: ambiguous argument '0d26d192d': unknown revision or path not in the working tree.
```

**Comment by gpanders** [comment]
@telemachus Try removing `opt.termguicolors = true`.

**Comment by telemachus** [comment]
> Try removing `opt.termguicolors = true`.

No joy. I commented out both `opt.termguicolors = true` and `opt.background = "light"`, but I get the same black flash.

**Pull request merged**